packages/app/src/containers/Canvas.tsx

```diff
@@ -249,6 +249,13 @@ export default memo(function Canvas() {
           event.preventDefault();
           break;
         }
+        case 'editPath': {
```

Comment:
We should probably pass the layerId + index of the target point to `"selectPoint"`, rather than its coordinates. That'll let us re-use the action elsewhere more easily, where we might not have a mouse coordinate. We also want to be more precise once we're in the reducer, since a coordinate might correspond to multiple layer points. And it'll be consistent with how we handle other selections (e.g. `"selectLayer"`). 

For the types, instead of `SetNumberMode`, we should use `SelectionType` like `"selectLayer"` uses:

https://github.com/noya-app/noya/blob/5ffef92f08def4aff8d6fd4e3cdd1dd2e9d01f2c/packages/noya-state/src/reducers/layerReducer.ts#L38-L42

This already has a helper function `updateSelection` for modifying selection arrays. We can call this for each layer in the `selectedPointLists`:

https://github.com/noya-app/noya/blob/5ffef92f08def4aff8d6fd4e3cdd1dd2e9d01f2c/packages/noya-state/src/reducers/layerReducer.ts#L264

For the selection logic, I think we can follow a similar pattern to selecting layers:

https://github.com/noya-app/noya/blob/5ffef92f08def4aff8d6fd4e3cdd1dd2e9d01f2c/packages/app/src/containers/Canvas.tsx#L261-L279

So supposing we made a new type that uniquely identifies a selected point:

```ts
type SelectedPoint = [layerId: string, index: number]
```

Then we would first figure out which point is selected, moving the bounding rect stuff that's in the reducer into `Canvas.tsx`, and afterward we could dispatch `"selectPoint"` actions something like this:

```ts
// Our selected point if we clicked one, or undefined
const selectedPoint: SelectedPoint | undefined = [layer.do_objectID, index]

if (selectedPoint) {
  // Check if the point is already in the selection array
  if (draft.selectedPointLists[layer.do_objectID]?.includes(index)) {
    // If shift is held and more than 1 point is selected
    if (event.shiftKey && Object.values(draft.selectedPointLists).flat().length !== 1) {
      dispatch('selectPoint', selectedPoint, 'difference');
    }
  } else {
    dispatch('selectPoint', selectedPoint, event.shiftKey ? 'intersection' : 'replace');
  }
} else {
  dispatch('selectPoint', undefined);
}
```

^ I tried to adapt the `"selectLayer"` logic so that the behavior is the same

Reply:
Type `SetNumberMode` changed to `SelectionType` in pointReducer. "adjust" changed to "difference" (I'm not sure if that correct)